### PR TITLE
Show tables in database with owner policy

### PR DESCRIPTION
**Pull request by saihemanth-cloudera**
### What changes were proposed in this pull request?

Owner of the tables in DB will be able to see the tables while issuing the command "Show tables in database" without the need of ranger policy.

### Why are the changes needed?

Owner of the tables should be able to see the tables.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

From spark-shell client and add a unit test case.

### CODE REVIEW

File: standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStore.java [hunk 1]
```diff
@@ -5695,8 +5695,16 @@ private void alter_table_core(String catName, String dbname, String name, Table
...
-            parsedDbName[CAT_NAME], parsedDbName[DB_NAME], ret);
+        if(ret !=  null && !ret.isEmpty()) {
+          List<Table> tableInfo = new ArrayList<>();
+          tableInfo = getMS().getTableObjectsByName(parsedDbName[CAT_NAME], parsedDbName[DB_NAME], ret);
+          tableInfo = FilterUtils.filterTablesIfEnabled(isServerFilterEnabled, filterHook, tableInfo);
+          List<String> result = new ArrayList<>(); // Cannot do ret.clear() because of query cannot be modified error in tests.
+          for (Table tbl : tableInfo) {
+            result.add(tbl.getTableName());
+          }
+          return result;
```
Comment (yongzhi) [inline standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStore.java:5706]: Better follow the same pattern as the original function, return at the end.

File: standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStore.java [hunk 2]
```diff
@@ -5695,8 +5695,16 @@ private void alter_table_core(String catName, String dbname, String name, Table
       String[] parsedDbName = parseDbName(dbname, conf);
       try {
         ret = getMS().getTables(parsedDbName[CAT_NAME], parsedDbName[DB_NAME], pattern);
-        ret = FilterUtils.filterTableNamesIfEnabled(isServerFilterEnabled, filterHook,
-            parsedDbName[CAT_NAME], parsedDbName[DB_NAME], ret);
+        if(ret !=  null && !ret.isEmpty()) {
+          List<Table> tableInfo = new ArrayList<>();
+          tableInfo = getMS().getTableObjectsByName(parsedDbName[CAT_NAME], parsedDbName[DB_NAME], ret);
+          tableInfo = FilterUtils.filterTablesIfEnabled(isServerFilterEnabled, filterHook, tableInfo);
```
Comment (yongzhi) [inline standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStore.java:5701]: Please add comments on why you pass-in tableInfo objects

File: standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStore.java [hunk 3]
```diff
@@ -5695,8 +5695,16 @@ private void alter_table_core(String catName, String dbname, String name, Table
...
+          List<String> result = new ArrayList<>(); // Cannot do ret.clear() because of query cannot be modified error in tests.
```
Comment (yongzhi) [inline standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStore.java:5702]: Not quite understand the comment.  Maybe ret = new ArrayList<>();   good enough?

File: itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/security/authorization/plugin/TestHiveAuthorizerCheckInvocation.java [hunk 4]
```diff
@@ -600,6 +600,22 @@ public void testReplDump() throws Exception {
...
+    final String tableName2 = "table2";
+    driver.run("create table " + dbName+"."+tableName2 + "(eid int, ecode int)");
+    reset(mockedAuthorizer);
+
+    int status = driver.compile("show tables in "+dbName, true);
+    assertEquals(0, status);
+    Pair<List<HivePrivilegeObject>, List<HivePrivilegeObject>> io = getHivePrivilegeObjectInputs();
+    List<HivePrivilegeObject> inputs = io.getLeft();
+    HivePrivilegeObject dbObj = inputs.get(0);
+    assertEquals("input type", HivePrivilegeObjectType.DATABASE, dbObj.getType());
```
Comment (yongzhi) [inline itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/security/authorization/plugin/TestHiveAuthorizerCheckInvocation.java:616]: Is that possible also to assert on owner related values?

Comment (saihemanth-cloudera) [reply itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/security/authorization/plugin/TestHiveAuthorizerCheckInvocation.java:635]: I have added the assert on owner info.

**Review by yongzhi — APPROVED** [review]
LGTM

**Review by yongzhi — APPROVED** [review]

**Pull request merged**